architectds/architectds.py

```diff
@@ -1052,6 +1052,57 @@ def add_ptexconv_tex4x4(self, in_dirs, out_dir='ptexconv'):
             self.add_data_file(out_path_tex, out_path_dir)
             self.add_data_file(out_path_idx, out_path_dir)
             self.add_data_file(out_path_pal, out_path_dir)
+    
+    def add_ptexconv(self, in_dirs, out_dir='ptexconv'):
+        '''
+        This function gets as input a list of directories. It will look for
+        files with extension '.png' and '.jpg'. It will then look for files alongside
+        them with the same name but with the extension '.ptxc'. The, it will create
+        rules to convert them to an NDS format with ptexconv using the arguments supplied
+        in the .ptxc file and add them to the CPU as binary data.
+        '''
+        full_out_dir = os.path.join(self.out_assets_path, out_dir)
+
+        in_out_files = []
+
+        for in_dir in in_dirs:
+            in_files = gen_input_file_list(in_dir, ('.png'))
+            in_out_files.extend(gen_out_file_list(in_files, in_dir, full_out_dir, '.png', '_png'))
+            in_files = gen_input_file_list(in_dir, ('.jpg'))
+            in_out_files.extend(gen_out_file_list(in_files, in_dir, full_out_dir, '.jpg', '_jpg'))
+            
+        for in_out_file in in_out_files:
+            ptexconv_out_path = in_out_file.out_path
+
+            out_path_dir = get_parent_dir(ptexconv_out_path)
+            self.add_dir_target(out_path_dir)
+
+            in_path_png = in_out_file.in_path
+            in_args = ''
+            with open(f'{os.path.splitext(in_out_file.in_path)[0]}.ptxc', 'r') as arg_file:
+                in_args = arg_file.readline()
```

Comment:
So the only thing I'd change is to support comments in ptxc files like in the case of grit files.

Reply:
Yes, that makes sense. Args should still be a one liner, so a while statement reading until we find a line that doesn't start with `#` when stripped should do the trick.